cpp/src/binaryop/binaryop.cpp

```diff
@@ -571,26 +626,51 @@ std::unique_ptr<column> fixed_point_binary_operation(column_view const& lhs,
 std::unique_ptr<column> fixed_point_binary_operation(column_view const& lhs,
                                                      column_view const& rhs,
                                                      binary_operator op,
+                                                     thrust::optional<cudf::data_type> output_type,
                                                      rmm::cuda_stream_view stream,
                                                      rmm::mr::device_memory_resource* mr)
 {
   using namespace numeric;
 
   CUDF_EXPECTS(is_supported_fixed_point_binop(op), "Unsupported fixed_point binary operation");
-  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(),
-               "Both columns must be of the same fixed_point type");
+  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(), "Data type mismatch");
+  if (op == binary_operator::TRUE_DIV)
+    CUDF_EXPECTS(output_type.has_value(), "TRUE_DIV requires result_type.");
 
-  auto const scale       = compute_scale_for_binop(op, lhs.type().scale(), rhs.type().scale());
-  auto const output_type = is_comparison_binop(op) ? data_type{type_id::BOOL8}  //
-                                                   : data_type{lhs.type().id(), scale};
-  auto out = make_fixed_width_column_for_output(lhs, rhs, op, output_type, stream, mr);
+  auto const scale = op == binary_operator::TRUE_DIV
+                       ? output_type.value().scale()
```

Comment:
Does this mean that the `output_type` has to be `fixed_point`? Or are you relying on the zero initialization of scale in dtype. If former, then there's a missing EXPECTS, if latter then it's a bit of a readability issue but not mandatory to fix.

Reply:
Good point. It has to be `fixed_point`. Will add now.